packages/ember-glimmer/lib/component.js

```diff
@@ -71,18 +71,16 @@ const Component = CoreView.extend(
       let args, reference;
 
       if ((args = this[ARGS]) && (reference = args[key])) {
-        if (reference[UPDATE]) {
-          reference[UPDATE](get(this, key));
-        } else {
-          let name = this._debugContainerKey.split(':')[1];
-          let value = get(this, key);
-          throw new Error(strip`
+        let name = this._debugContainerKey.split(':')[1];
```

Comment:
This is only used for the assertion, but as written it will not get stripped.  Can you refactor to avoid splitting in prod builds where the assertion is not needed?  Something like:

```js
let value = get(this, key);
runInDebug(function() {
  let name = this._debugContainerKey.split(':')[1];
  assert(strip`
    Cannot set the \`${key}\` property (on component ${name}) to
    \`${value}\`. The \`${key}\` property came from an immutable
    binding in the template, such as {{${name} ${key}="string"}}
    or {{${name} ${key}=(if theTruth "truth" "false")}}.
  `, reference[UPDATE]);
});

reference[UPDATE](value);
```